### PR TITLE
Added runAsUser configuration property.

### PR DESCRIPTION
**Pull request by razvan**
**IMPORTANT** Currently the Stackable agent doesn't support writable volumes so services that need those will not be able to run as non-root.

## Description

Added `runAsUser` configuration property to the service configuration. If present, it will set the desired user name in the pod's security context.

Part of https://github.com/stackabletech/issues/issues/59

## Review Checklist
- [ ] Code contains useful comments
- [ ] (Integration-)Test cases added (or not applicable)
- [ ] Documentation added (or not applicable)
- [ ] Changelog updated (or not applicable)

### CODE REVIEW

**Comment by razvan** [comment]
One problem with this seems to be that the `ConfigMap` files belong to `root`:
```
root@f1fb00307a4b:/agent# ls -l /tmp/zookeeper/
total 4
-rw-r--r-- 1 root root 1 Aug 24 15:47 myid
root@f1fb00307a4b:/agent# ls -l /tmp/ | grep zookeeper
drwxr-xr-x 2 root      root      4096 Aug 24 15:47 zookeeper
root@f1fb00307a4b:/agent# 
```

and therefore the service cannot start:

```
Aug 24 15:47:11 f1fb00307a4b zkServer.sh[710]: 2021-08-24 15:47:11,795 [myid:1] - ERROR [main:ZooKeeperServerMain@75] - Unable to access datadir, exiting abnormally
Aug 24 15:47:11 f1fb00307a4b zkServer.sh[710]: org.apache.zookeeper.server.persistence.FileTxnSnapLog$DatadirException: Unable to create data directory /tmp/zookeeper/version-2
``

File: deploy/config-spec/properties.yaml [hunk 1]
```diff
@@ -107,6 +107,24 @@ properties:
...
+  - property: &zooLogDir
+      propertyNames:
+        - name: "ZOO_LOG_DIR"
+          kind:
+            type: "env"
+      datatype:
+        type: "string"
+        unit: *unitDirectory
+      defaultValues:
+        - value: "/var/log/stackable/servicelogs"
```
Comment (lfrancke) [inline deploy/config-spec/properties.yaml:119]: Hmmm is this correct?
Doesn't this also need a zookeeper subdirectory?

File: deploy/config-spec/properties.yaml [hunk 2]
```diff
@@ -107,6 +107,24 @@ properties:
...
+      recommendedValues:
+        - value: "/var/log/stackable/servicelogs"
+      roles:
+        - name: "server"
+          required: true
+      asOfVersion: "0.0.0"
+      description: "The port where ZooKeeper metrics are exposed as a Prometheus endpoint."
```
Comment (lfrancke) [inline deploy/config-spec/properties.yaml:126]: Wrong docs

File: operator/src/lib.rs [hunk 3]
```diff
@@ -891,6 +898,12 @@ impl ZookeeperState {
             );
         }
 
+        let security_context = run_as_user.map_or(PodSecurityContextBuilder::new().build(), |u| {
```
Comment (lfrancke) [inline operator/src/lib.rs:901]: Is the default correct for us? Before we had none here and now we use a default one. Is that a difference?

**Review by lfrancke — CHANGES_REQUESTED** [review]

**Comment by razvan** [comment]
Since this functionality is not supported by the Stackable agent I will close this PR but keep the branch for future reference.

**Pull request closed without merging**